File: src/my_cpp_pkg/src/robot_news_station.cpp
#include <stdio.h>
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

// Creating C++ Node with OOP
class RobotNewsStationNode: public rclcpp::Node
{
    public:
        RobotNewsStationNode(): Node("robot_news_station"), robot_name_("R2D2")
        {
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
            timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                             std::bind(&RobotNewsStationNode::publishNews, this));

            RCLCPP_INFO(this->get_logger(), "Robot News Station has been started.");
        };
    
    private:

        void publishNews()
        {
            auto msg = example_interfaces::msg::String();
            msg.data = std::string("Hi, this is ") + robot_name_ + std::string(" from the Robot News Station");
            publisher_->publish(msg);
        }

        std::string robot_name_;
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv){

    // Initialize ROS2 communication
    rclcpp::init(argc, argv);

    // Object created into Shared pointer 
    // auto node = std::make_shared<rclcpp::Node>("cpp_test");

    // RCLCPP_INFO(node->get_logger(), "Hello CPP Node");

    auto node = std::make_shared<RobotNewsStationNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

# inheriting from Node class
class RobotNewsStationNode(Node):

    def __init__(self):
        super().__init__("robot_news_station")

        self.robot_name_ = "TAC"
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.timer_ = self.create_timer(0.5, self.publish_new)
        self.get_logger().info("Robot News Station has been started")

    def publish_new(self):
        msg = String()
        msg.data = "Hi, this is " + str(self.robot_name_) + " from the robot news station." 
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)  # starts ROS2 communication and other features

    # node = Node("py_test")  # first node created
    # node.get_logger().info("Hello ROS2") 
    node = RobotNewsStationNode()

    rclpy.spin(node)  # node will be spinning and program wont stop
    rclpy.shutdown() # shutting down opened ros2 communication

if __name__ == "__main__":
    main()